portal-impl/src/com/liferay/portal/service/impl/CountryLocalServiceImpl.java

```diff
@@ -210,6 +224,23 @@ public Country getCountryByNumber(long companyId, String number)
 		return countryPersistence.findByC_Number(companyId, number);
 	}
 
+	@Override
+	public List<Country> searchCountries(
+			long companyId, Boolean active, String keywords, int start, int end,
+			OrderByComparator<Country> orderByComparator)
+		throws PortalException {
+
+		return dslQuery(
+			_getGroupByStep(
+				DSLQueryFactoryUtil.selectDistinct(CountryTable.INSTANCE),
+				companyId, active, keywords
+			).orderBy(
+				CountryTable.INSTANCE, orderByComparator
+			).limit(
+				start, end
+			));
+	}
+
```

Comment:
We also need to include a `searchCount` method as well. It's almost identical to this, but with a different from statement and no limit. There should be an example in the same class where you got the DSL query from.